subiquity/models/filesystem.py

```diff
@@ -646,3 +69,599 @@ def _dehumanize_size(size):
+# Disk, Partition etc classes correspond to entries in curtin's
+# storage config list. They are mostly 'dumb data', all the logic is
+# in the FilesystemModel or FilesystemController classes.
+
+
+@attr.s
+class Disk:
+    id = attr.ib(default=id_factory("disk"))
+    type = attr.ib(default="disk")
+    ptable = attr.ib(default='gpt')
+    serial = attr.ib(default=None)
+    path = attr.ib(default=None)
+    model = attr.ib(default=None)
+    wipe = attr.ib(default=None)
+    preserve = attr.ib(default=False)
+    name = attr.ib(default="")
+    grub_device = attr.ib(default=False)
+
+    _partitions = attr.ib(default=attr.Factory(list), repr=False) # [Partition]
+    _fs = attr.ib(default=None, repr=False) # Filesystem
+    _info = attr.ib(default=None)
+
+    @classmethod
+    def from_info(self, info):
+        d = Disk(info=info)
+        d.serial = info.serial
+        d.path = info.name
+        d.model = info.model
+        return d
+
+    @property
+    def available(self):
+        return self.used < self.size
+
+    @property
+    def next_partnum(self):
+        return len(self._partitions) + 1
+
+    @property
+    def size(self):
+        return self._info.size
+
+    @property
+    def used(self):
+        if self._fs is not None:
+            return self.size
+        r = 0
+        for p in self._partitions:
+            r += p.size
+        return r
+
+    @property
+    def free(self):
+        return self.size - self.used
+
+
+@attr.s
+class Partition:
+    id = attr.ib(default=id_factory("part"))
+    type = attr.ib(default="partition")
+    number = attr.ib(default=0)
+    device = attr.ib(default=None) # Disk
+    size = attr.ib(default=None)
+    wipe = attr.ib(default=None)
+    flag = attr.ib(default=None)
+    preserve = attr.ib(default=False)
+
+    _fs = attr.ib(default=None, repr=False) # Filesystem
+
+    @property
+    def available(self):
+        if self.flag == 'bios_grub':
+            return False
+        if self._fs is None:
+            return True
+        if self._fs._mount is None:
+            fs_obj = FilesystemModel.fs_by_name[self._fs.fstype]
+            return fs_obj.is_mounted
+        return False
+
+    @property
+    def path(self):
+        return "%s%s"%(self.device.path, self.number)
+
+
+@attr.s
+class Filesystem:
+    id = attr.ib(default=id_factory("fs"))
+    type = attr.ib(default="format")
+    fstype = attr.ib(default=None)
+    volume = attr.ib(default=None) # Partition or Disk
+    label = attr.ib(default=None)
+    uuid = attr.ib(default=None)
+    preserve = attr.ib(default=False)
+    _mount = attr.ib(default=None, repr=False) # Mount
+
+
+@attr.s
+class Mount:
+    id = attr.ib(default=id_factory("mount"))
+    type = attr.ib(default="mount")
+    device = attr.ib(default=None) # Filesystem
+    path = attr.ib(default=None)
+
+
+def align_up(size, block_size=1 << 20):
+    return (size + block_size - 1) & ~(block_size - 1)
+
+GPT_END_RESERVE = 1 << 20
```

Comment:
You could move this above `align_up()` and use it as the default value for `block_size`.